locust/stats.py

```diff
@@ -366,6 +455,17 @@ def percentile(self, tpl=" %-" + str(STATS_NAME_WIDTH) + "s %8d %6d %6d %6d %6d
             self.get_response_time_percentile(0.99),
             self.max_response_time
         )
+    
+    def _cache_response_times(self, t):
+        self.response_times_cache[t] = CachedResponseTimes(
+            response_times=copy(self.response_times),
+            num_requests=self.num_requests,
+        )
+        if len(self.response_times_cache) > 20:
```

Comment:
Since the cache might not hold response times for every second. We scan through the cache at time()-CURRENT_RESPONSE_TIME_PERCENTILE_WINDOW +/- 10 seconds to find a cache entry which we'll use to calculate the delta in order to get the "current" response time percentile.

I've changed the cache size to be CURRENT_RESPONSE_TIME_PERCENTILE_WINDOW + 10.

I guess we *could* make the number of seconds we scan configurable as well, but I really don't see a use-case where one would need to change this.